test/mpg/core_test.clj

```diff
@@ -2,17 +2,43 @@
   (:require [clojure.test :refer :all]
             [clojure.java.jdbc :as sql]
             [environ.core :refer [env]]
-            [mpg.core :as mpg])
-  (:import [java.time LocalDate ZonedDateTime ZoneId]
+            [mpg.core :as mpg]
+            [mpg.data :as d]
+            [mpg.datetime :as dt])
+  (:import [java.nio ByteBuffer]
+           [java.sql Timestamp]
+           [java.time LocalDate ZonedDateTime ZoneId]
            [org.postgresql.util PGobject]))
 
 (mpg/patch)
 
+(defn to-byte-string [thing]
+  (cond (instance? (Class/forName "[B") thing) (String. ^bytes thing)
+        (instance? ByteBuffer thing) (String. ^bytes (d/bytebuf->array thing))
+        :else (-> (str "WTF is this? " thing " " (type thing))
+                  (ex-info {:got thing})
+                  throw)))
+
+(defn to-ts [thing]
+  (cond (instance? Timestamp thing) thing
+        (instance? java.util.Date thing) (Timestamp. (.getTime ^java.util.Date thing))
+        (instance? ZonedDateTime thing) (dt/zoneddatetime->timestamp thing)
+        :else (-> (str "WTF is this? " thing " " (type thing))
+                  (ex-info {:got thing})
+                  throw)))
+
+(defn to-date [thing]
+  (cond (instance? java.util.Date thing) (dt/truncate-date thing)
+        (instance? LocalDate thing) (dt/localdate->date thing)
+        :else (-> (str "WTF is this? " thing " " (type thing))
+                  (ex-info {:got thing})
+                  throw)))
+
 ;; Allow the user to specify connection parameters on the command line
 (def pg
   (as-> {:subprotocol "postgresql"
          :subname (or (env :mpg-test-db-uri)
-                      "//127.0.0.1:5432/sanity_test")} $
+                      "//127.0.0.1:5432/mpg_test")} $
```

Comment:
Right, so I had a thinko with Travis. Last time, I didn't have permission to, but you writing this message reminded me that actually for this, all I have to do is prod the travis.yml and commit again, so I'm working on getting the build working again.